Gems/Compression/Code/Include/Compression/CompressionInterfaceAPI.h

```diff
@@ -70,28 +69,44 @@ namespace Compression
+
+
+        //! Registers compression interface, but does not take ownership of it
         //! If a compression interface with a CompressionAlgorithmId is registered that
         //! matches the input compression interface, then registration does not occur
-        //! and the unique_ptr refererence is unmodified and can be re-used by the caller
         //!
-        //! @param compressionInterface unique pointer to compression interface to register
-        //! @return true if the ICompressionInterface was successfully registered, otherwise false
-        virtual bool RegisterCompressionInterface(AZStd::unique_ptr<ICompressionInterface>&&) = 0;
+        //! Registers compression interface, but does not take ownership of it
+        //! @param algorithmId Unique id to associate with compression interface
+        //! @param compressionInterface compression interface to register
+        //! @return true if the ICompressionInterface was successfully registered
+        virtual bool RegisterCompressionInterface(CompressionAlgorithmId algorithmId, ICompressionInterface&) = 0;
 
-        //! Unregisters the Compression Interface with the specified Id if it is registered with the factory
+        //! Unregisters the compression interface with the specified id
         //!
-        //! @param compressionAlgorithmId unique Id that identifies the Compression Interface
-        //! @return true if the ICompressionInterface was unregistered, otherwise false
-        virtual bool UnregisterCompressionInterface(CompressionAlgorithmId) = 0;
-
-        //! Queries the Compression Interface with the compression algorithmd Id
-        //! @param compressionAlgorithmId unique Id of Compression Interface to query
-        //! @return pointer to the Compression Interface or nullptr if not found
-        virtual ICompressionInterface* FindCompressionInterface(CompressionAlgorithmId) const = 0;
+        //! @param compressionAlgorithmId unique Id that identifies the compression interface
+        //! @return true if the unregistration is successful
+        virtual bool UnregisterCompressionInterface(CompressionAlgorithmId algorithmId) = 0;
+
+        //! Queries the compression interface with the compression algorithmd Id
+        //! @param compressionAlgorithmId unique Id of compression interface to query
+        //! @return pointer to the compression interface or nullptr if not found
+        [[nodiscard]] virtual ICompressionInterface* FindCompressionInterface(CompressionAlgorithmId) const = 0;
+
+
+        //! Return true if there is an compression interface registered with the specified id
+        //! @param algorithmId CompressionAlgorithmId to determine if an compression interface is registered
+        //! @return bool indicating if there is an compression interface with the id registered
+        [[nodiscard]] virtual bool IsRegistered(CompressionAlgorithmId) const = 0;
```

Comment:
Should match the doxygen entry for ```algorithmId```
```suggestion
        [[nodiscard]] virtual bool IsRegistered(CompressionAlgorithmId algorithmId ) const = 0;
```